kernel/memory_x86_64/src/lib.rs

```diff
@@ -110,19 +110,19 @@ where
         // The linker script (linker_higher_half.ld) defines the following order of sections:
         // |------|-------------------|-------------------------------|
         // | Sec  |    Sec Name       |    Description / purpose      |
-        // | Num  |                   |                               |
+        // | Idx  |                   |                               |
         // |------|---------------------------------------------------|
-        // | (1)  | .init             | start of executable pages     |
-        // | (2)  | .text             | end of executable pages       |
-        // | (3)  | .rodata           | start of read-only pages      |
-        // | (4)  | .eh_frame         | part of read-only pages       |
-        // | (5)  | .gcc_except_table | part/end of read-only pages   |
-        // | (6)  | .tdata            | part/end of read-only pages   |
-        // | (7)  | .tbss             | part/end of read-only pages   |
-        // | (8)  | .data             | start of read-write pages     | 
-        // | (9)  | .bss              | end of read-write pages       |
-        // | (10) | .page_table       | separate .data-like section   |
-        // | (11) | .stack            | separate .data-like section   |
+        // | (0)  | .init             | start of executable pages     |
+        // | (1)  | .text             | end of executable pages       |
+        // | (2)  | .rodata           | start of read-only pages      |
+        // | (3)  | .eh_frame         | part of read-only pages       |
+        // | (4)  | .gcc_except_table | part/end of read-only pages   |
+        // | (5)  | .tdata            | part/end of read-only pages   |
+        // | (6)  | .tbss             | part/end of read-only pages   |
+        // | (7)  | .data             | start of read-write pages     | 
+        // | (8)  | .bss              | end of read-write pages       |
+        // | (9)  | .page_table       | separate .data-like section   |
+        // | (10) | .stack            | separate .data-like section   |
```

Comment:
I noticed you updated these shndxs -- those are supposed to match what is shown in the ELF file. I don't think `0` is a valid one, so kindly update those to match the output of `readelf`... unless I'm mis-remembering

Reply:
I changed that so that they match the output of objdump (zero-based); I'll revert that